### PR TITLE
Auto delete tables

### PR DESCRIPTION
**Pull request by ohadbitt**
**Features:**
- Use policy Auto delete instead of checking occasionally for lost tables

### CODE REVIEW

File: connector/src/main/scala/com/microsoft/kusto/spark/utils/KustoClient.scala [hunk 1]
```diff
@@ -66,12 +70,17 @@ class KustoClient(val clusterAlias: String, val engineKcsb: ConnectionStringBuil
...
 
-    // Create a temporary table with the kusto or dataframe parsed schema with 1 day retention
+    // Create a temporary table with the kusto or dataframe parsed schema with retention and delete set to after the
+    // write operation times out
     engineClient.execute(database, generateTempTableCreateCommand(tmpTableName, tmpTableSchema))
-    engineClient.execute(database, generateTableAlterRetentionPolicy(tmpTableName, "001:00:00:00", recoverable = false))
+    engineClient.execute(database, generateTableAlterRetentionPolicy(tmpTableName,
+      DurationFormatUtils.formatDuration(writeTimeout.toMillis, durationFormat,true),
+      recoverable=false))
+    val instant = Instant.now.plusSeconds(writeTimeout.toSeconds.intValue())
```
Comment (yihezkel) [inline connector/src/main/scala/com/microsoft/kusto/spark/utils/KustoClient.scala:82]: nit: plusSeconds accepts a long, so might as well use "writeTimeout.toSeconds.longValue()"

File: connector/src/main/scala/com/microsoft/kusto/spark/utils/CslCommandsGenerator.scala [hunk 2]
```diff
@@ -148,6 +133,10 @@ private[kusto] object CslCommandsGenerator {
     s""".alter table ${KustoQueryUtils.normalizeTableName(tmpTableName)} policy retention '{ "SoftDeletePeriod": "$period", "Recoverability":"${if (recoverable) "Enabled" else "Disabled"}" }' """
   }
 
+  def generateTableAlterAutoDeletePolicy(tmpTableName: String, expiryDate: String): String = {
```
Comment (yihezkel) [inline connector/src/main/scala/com/microsoft/kusto/spark/utils/CslCommandsGenerator.scala:136]: Why is the ExpiryDate (type of DateTime) parameter a String?

**Review by yihezkel — APPROVED** [review]
Minor comments

File: connector/src/main/scala/com/microsoft/kusto/spark/utils/CslCommandsGenerator.scala (hunk 2 above)
Comment (ohadbitt) [reply connector/src/main/scala/com/microsoft/kusto/spark/utils/CslCommandsGenerator.scala:136]: will change

**Pull request merged**